python/pyhail/dataset.py

```diff
@@ -956,10 +1050,43 @@ def filter_alleles(self, condition, annotation=None, subset=True, keep=True, fil
         return self.hc.run_command(self, pargs)
 
     def filter_genotypes(self, condition, keep=True):
-        """Filter variants based on expression.
+        """Filter genotypes based on expression.
+
+        **Examples**
+
+        Filter genotypes by allele balance dependent on genotype call
+
+        >>> (vds.filter_genotypes('let ab = g.ad[1] / g.ad.sum in'
+        >>>                      '((g.isHomRef && ab <= 0.1) ||'
+        >>>                      '(g.isHet && ab >= 0.25 && ab <= 0.75) ||'
+        >>>                      '(g.isHomVar && ab >= 0.9))'))
+
+        **Implementation Details**
+
+        Use the `Hail expression language <../reference.html#HailExpressionLanguage>`_ to supply a ``condition`` (boolean expression) involving the following exposed data structures:
+
+        - ``s`` (*Sample*): :ref:`sample`
+        - ``v`` (*Variant*): :ref:`variant`
+        - ``sa``: sample annotations
+        - ``va``: variant annotations
+        - ``global``: global annotations
+
+        All variables and values are case sensitive.
```

Comment:
This appears in a number of places.  Should we do that, or just make this remark once in the Hail expression language section?